homeassistant/components/sensor/sabnzbd.py

```diff
@@ -4,216 +4,72 @@
-                save_json(hass.config.path(CONFIG_FILE), conf)
-                req_config = _CONFIGURING.pop(host)
-                configurator.async_request_done(req_config)
-
-            hass.async_add_job(success)
-
-    _CONFIGURING[host] = configurator.async_request_config(
-        DEFAULT_NAME,
-        async_configuration_callback,
-        description='Enter the API Key',
-        submit_caption='Confirm',
-        fields=[{'id': 'api_key', 'name': 'API Key', 'type': ''}]
-    )
-
-
-@asyncio.coroutine
-def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
-    """Set up the SABnzbd platform."""
-    from pysabnzbd import SabnzbdApi
-
-    if discovery_info is not None:
-        host = discovery_info.get(CONF_HOST)
-        port = discovery_info.get(CONF_PORT)
-        name = DEFAULT_NAME
-        use_ssl = discovery_info.get('properties', {}).get('https', '0') == '1'
-    else:
-        host = config.get(CONF_HOST)
-        port = config.get(CONF_PORT)
-        name = config.get(CONF_NAME, DEFAULT_NAME)
-        use_ssl = config.get(CONF_SSL)
-
-    api_key = config.get(CONF_API_KEY)
-
-    uri_scheme = 'https://' if use_ssl else 'http://'
-    base_url = "{}{}:{}/".format(uri_scheme, host, port)
-
-    if not api_key:
-        conf = load_json(hass.config.path(CONFIG_FILE))
-        if conf.get(base_url, {}).get('api_key'):
-            api_key = conf[base_url]['api_key']
-
-    if not (yield from async_check_sabnzbd(SabnzbdApi, base_url, api_key)):
-        request_configuration(base_url, name, hass, config,
-                              async_add_devices, SabnzbdApi)
-        return
-
-    setup_sabnzbd(base_url, api_key, name, config,
-                  async_add_devices, SabnzbdApi)
+
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
+    """Set up the SABnzbd sensors."""
+    sab_api_data = hass.data[DATA_SABNZBD]
+    sensors = sab_api_data.sensors
+    client_name = sab_api_data.name
+    async_add_devices([SabnzbdSensor(hass, sensor, sab_api_data, client_name)
+                       for sensor in sensors])
 
 
 class SabnzbdSensor(Entity):
     """Representation of an SABnzbd sensor."""
 
-    def __init__(self, sensor_type, sabnzbd_api, client_name):
+    def __init__(self, hass, sensor_type, sabnzbd_api_data, client_name):
         """Initialize the sensor."""
+        self.hass = hass
+
+        self._client_name = client_name
+        self._field_name = SENSOR_TYPES[sensor_type][2]
         self._name = SENSOR_TYPES[sensor_type][0]
-        self.sabnzbd_api = sabnzbd_api
-        self.type = sensor_type
-        self.client_name = client_name
+        self._sabnzbd_api = sabnzbd_api_data
         self._state = None
+        self._type = sensor_type
         self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
 
+        async_dispatcher_connect(hass, SIGNAL_SABNZBD_UPDATED,
```

Comment:
Move this to `async_added_to_hass`. Calling the the state update before the entity has been added to home assistant will error.